### ISSUE TITLE
Cmake limit under travis

### ISSUE THREAD

**Issue by Darelbi** [Darelbi/Infectorpp#9]
Travis (the build bot I use for continuos integration) is actually flawed (it has just an old Cmake version).

The build scripts should be refactored to use old Cmake version

**Comment by Darelbi** [comment]
to be more precise, I'm able to compile Infectorpp with a variety of clang and gcc versions on windows and linux using cmake (it also compiles out of the box by adding source to your project). But it requires latest cmake version. Still to verify if it compile on windows but microsoft did not made available on win xp/vista VS 2013 >.<

**Comment by onqtam** [comment]
cmake 2.8 is not that new. Here is how to get 2.8.12 on travis.ci:

```
before_install:
  - sudo add-apt-repository ppa:kubuntu-ppa/backports -y
  - sudo apt-get update -qq
  - sudo apt-get install --only-upgrade cmake
```

**Comment by Darelbi** [comment]
thanks but, it still doesn't work:

CMake Error at cmake/AddLibraryType.cmake:13 (target_compile_definitions):
  Unknown CMake command "target_compile_definitions".
Call Stack (most recent call first):
  CMakeLists.txt:22 (add_library_type)

**Comment by onqtam** [comment]
I'm looking at this
https://travis-ci.org/Darelbi/Infectorpp/jobs/67811037
and this
https://travis-ci.org/Darelbi/Infectorpp/jobs/67811038
and I don't see any cmake errors. I don't know.

Have you heard about this https://github.com/krzysztof-jusiak/di ?

**Comment by Darelbi** [comment]
Ah thanks, it seems to works now (probably travis needed some time to fecth the repo).

About the boost's proposal I'm not happy with it, It don't even mention infectorpp :D.

I know I started Infectorpp in 2013 as an experiment, but I think I pioneered that (I learnt about DI from a c# library, and to understand better the concept I tried immediatly to implement that, my noob code proven to be "right" in a certain sense and it was already C++ish enough at the time).

What I think is that Infectorpp was innovative (about few features, but also about few implementation details: also the first library using type erasure for dependency injection) a bit immature maybe but I'm still happy with it, any serious libray should mention it (just for the effort I put into that :P.. ok i'm exaggerating a bit mabye..)

The fact I used certain details is about that I wanted a clean as possible interface (interface that now is used in every post-infectorpp framework, if you take a look at pre-infectorpp frameworks they all have ugly interfaces, some even use XML >.<) so It is nice more people came to my same conclusions, but I'm bit unhappy they not looked around and saw some people already did something similiar (I mentioned from the first day the C# library that inspired me and I always stated "Inspired by that"). 

I planned the code for version 2 too much for not implementing it (also because Infectorpp2 is part of my thesis XD), and I think that it will stil be usefull regardless of what other frameworks can offer now.
